lib/MySQL_HostGroups_Manager.cpp

```diff
@@ -1437,7 +1437,13 @@ MySQL_Connection * MySrvConnList::get_random_MyConn(MySQL_Session *sess, bool ff
 	MySQL_Connection * conn=NULL;
 	unsigned int i;
 	unsigned int l=conns_length();
-	if (l && ff==false) {
+	bool needs_warming = false;
+	unsigned int total_connections = mysrvc->ConnectionsFree->conns_length()+mysrvc->ConnectionsUsed->conns_length();
+	unsigned int expected_warm_connections = mysql_thread___free_connections_pct*mysrvc->max_connections/100;
```

Comment:
@akshaysuryawanshi  I was reading through some of `shopify-proxysql` history recently, and I noticed that the fix in https://github.com/Shopify/shopify-proxysql/pull/34/commits/aaeff2fff7648ce7249295baef5749295bff6bf5 didn't make it here.

Reply:
@saunderst thank you for catching that. I have pushed the cleanup from Shopify patch to this branch.